test/PandoraInterface.cxx

```diff
@@ -336,14 +368,462 @@ void ProcessEvents(const Parameters &parameters, const Pandora *const pPrimaryPa
+        point2 = stop;
+    }
+    else if (!inStart && !inStop)
+    {
+        //std::cout << "Case 2: Start and end points are outside boundary" << std::endl;
+        bool ok = grid.intersect(ray, t0, t1);
+        if (ok)
+        {
+            point1 = ray.getPoint(t0);
+            point2 = ray.getPoint(t1);
+            //std::cout << "Boundary points: " << point1 << " and " << point2 << std::endl;
+        }
+        else
+        {
+            std::cout << "Voxel ray does not intersect boundary limits." << std::endl;
+            return currentVoxelList;
+        }
+    }
+    else if (inStart && !inStop)
+    {
+        //std::cout << "Case 3: Start inside boundary" << std::endl;
+        point1 = start;
+        bool ok = grid.intersect(ray, t0, t1);
+        if (ok)
+        {
+            point2 = ray.getPoint(t1);
+        }
+        else
+        {
+            std::cout << "Voxel ray does not intersect boundary" << std::endl;
+            return currentVoxelList;
+        }
+    }
+    else if (!inStart && inStop)
+    {
+        //std::cout << "Case 4: End inside boundary" << std::endl;
+        point2 = stop;
+        bool ok = grid.intersect(ray, t0, t1);
+        if (ok)
+        {
+            point1 = ray.getPoint(t0);
+        }
+        else
+        {
+            std::cout << "Voxel ray does not intersect boundary" << std::endl;
+            return currentVoxelList;
+        }
+    }
+
+    // Now create voxels between point1 and point2.
+    // Ray direction will be the same, but update starting point
+    ray.updateOrigin(point1);
+
+    const double epsilon(1e-3);
+    bool shuffle(true);
+
+    // Keep track of total voxel path length and energy (so far)
+    double totalPath(0.0), ETot(0.0);
+    int loop(0);
+
+    while (shuffle)
+    {
+
+        // Get point along path to define voxel bin (bottom corner)
+        pandora::CartesianVector voxelPoint = ray.getPoint(epsilon);
+
+        // Grid 3d bin containing this point; 4th element is the total bin number
+        std::array<long, 4> gridBins = grid.getBinIndices(voxelPoint);
+        long voxelID = gridBins[3];
+
+        long xBin = gridBins[0];
+        long yBin = gridBins[1];
+        long zBin = gridBins[2];
+
+        // Voxel bottom and top corners
+        const pandora::CartesianVector voxBot = grid.getPoint(xBin, yBin, zBin);
+        const pandora::CartesianVector voxTop = grid.getPoint(xBin + 1, yBin + 1, zBin + 1);
+
+        // Voxel box
+        const LArBox vBox(voxBot, voxTop);
+        // Get ray intersections with this box
+        bool result = vBox.intersect(ray, t0, t1);
+
+        if (!result)
+        {
+            shuffle = false;
+        }
+
+        // Voxel extent = intersection path difference
+        double dL(t1 - t0);
+        // For the first path length, use the distance from the
+        // starting ray point to the 2nd intersection t1
+        if (loop == 0)
+        {
+            dL = t1;
+        }
```

Comment:
Could simplify into a ternary operator? Though perhaps it would lose readability.

Reply:
Prefer to keep things as they are for readability.